docs/users.md

```diff
@@ -0,0 +1,41 @@
+We track different types of users in our browser code.
+
+There are three disjoint sets of users you care about
+for typical Zulip realms:
+
+- active users in your realm
+- cross-realm users like feedback@zulip.com
+- deactivated users
+
+You can also think in terms of these user populations:
+
+- current realm users (active users in your realm)
+- active users (adds cross-realm bots to current realm users)
+- all users (adds deactivated users to active users)
+
+Each of the above categories makes sense for certain UI functions.
+
+Let's start with the features that are restricted to active realm
+users, and let's end with features that can apply to all users.
+
+Only current realm users can...
+- be subscribed to public streams on your realm
+- be highlighted in the compose-fade feature
+- show up in your buddy list
+
+Active users can...
+- be sent PMs to
+- show up in your compose recipient typeaheads
+- show up in your at-mention typeaheads
+- show up in search suggestions
+
+All users can...
+- show up in your message stream
+- be narrowed to
+- be searched for (but not necessarily suggested)
+- can show up in your "Private Messages" sidebar
+
+We also have the mirroring world, where we have unknown users
+that we can send PMs to, and local-echo is allowed to create
+skeleton objects for them.
```

Comment:
We could clarify this to say something like "where users might send a message to a user who doesn't yet exist, and if they did so, the backend would query a third-party API and potentially create a "mirror dummy" user on-demand for that user account.".  Ultimately we'll want to add a section to our docs in general on mirror dummy users that this could link to, but that can be for later.